src/wmtk/EdgeMesh.cpp

```diff
@@ -0,0 +1,255 @@
+{
+    switch (type) {
+    case PrimitiveType::Vertex: {
+        return vertex_tuple_from_id(gid);
+    }
+    case PrimitiveType::Edge: {
+        return edge_tuple_from_id(gid);
+    }
+    case PrimitiveType::Face: {
+        throw std::runtime_error("no tet tuple supported for edgemesh");
+        break;
+    }
+    case PrimitiveType::Tetrahedron: {
+        throw std::runtime_error("no tet tuple supported for edgemesh");
+        break;
+    }
+    default: throw std::runtime_error("Invalid primitive type"); break;
+    }
+}
+
+Tuple EdgeMesh::vertex_tuple_from_id(long id) const
+{
+    ConstAccessor<long> ve_accessor = create_const_accessor<long>(m_ve_handle);
+    auto e = ve_accessor.index_access().scalar_attribute(id);
+    ConstAccessor<long> ev_accessor = create_const_accessor<long>(m_ev_handle);
+    auto ev = ev_accessor.index_access().vector_attribute(e);
+    for (long i = 0; i < 2; ++i) {
+        if (ev(i) == id) {
+            Tuple v_tuple = Tuple(i, -1, -1, e, get_cell_hash_slow(e));
+            return v_tuple;
+        }
+    }
+    throw std::runtime_error("vertex_tuple_from_id failed");
+}
+
+Tuple EdgeMesh::edge_tuple_from_id(long id) const
+{
+    Tuple e_tuple = Tuple(0, -1, -1, id, get_cell_hash_slow(id));
+
+    assert(is_valid_slow(e_tuple));
+    return e_tuple;
+}
+
+bool EdgeMesh::is_valid(const Tuple& tuple, ConstAccessor<long>& hash_accessor) const
+{
+    if (tuple.is_null()) return false;
+
+    if (tuple.m_local_vid < 0 || tuple.m_global_cid < 0) return false;
+
+    return Mesh::is_hash_valid(tuple, hash_accessor);
+}
+
+bool EdgeMesh::is_connectivity_valid() const
+{
+    // get accessors for topology
+    ConstAccessor<long> ev_accessor = create_const_accessor<long>(m_ev_handle);
+    ConstAccessor<long> ee_accessor = create_const_accessor<long>(m_ee_handle);
+    ConstAccessor<long> ve_accessor = create_const_accessor<long>(m_ve_handle);
+    ConstAccessor<char> v_flag_accessor = get_flag_accessor(PrimitiveType::Vertex);
+    ConstAccessor<char> e_flag_accessor = get_flag_accessor(PrimitiveType::Edge);
+
+    // VE and EV
+    for (long i = 0; i < capacity(PrimitiveType::Vertex); ++i) {
+        if (v_flag_accessor.index_access().scalar_attribute(i) == 0) {
+            wmtk::logger().debug("Vertex {} is deleted", i);
+            continue;
+        }
+        int cnt = 0;
+        for (long j = 0; j < 2; ++j) {
+            if (ev_accessor.index_access().vector_attribute(
+                    ve_accessor.index_access().scalar_attribute(i))[j] == i) {
+                cnt++;
+            }
+        }
+        if (cnt == 0) {
+            return false;
+        }
+    }
+
+    // EV and EE
+    for (long i = 0; i < capacity(PrimitiveType::Edge); ++i) {
+        if (e_flag_accessor.index_access().scalar_attribute(i) == 0) {
+            wmtk::logger().debug("Edge {} is deleted", i);
+            continue;
+        }
+        // TODO: need to handle cornor case (self-loop)
```

Comment:
You have a test case for that, right? Then this test case should not pass. You can flag it with `[.]` but it should contain a `REQUIRE(false);`.